Kafka/src/Akka.Streams.Kafka/Settings/Subscriptions.cs

```diff
@@ -3,49 +3,81 @@
 
 namespace Akka.Streams.Kafka.Settings
 {
-    public interface ISubscription { }
-    public interface IManualSubscription : ISubscription { }
-    public interface IAutoSubscription : ISubscription { }
+    public delegate void TopicPartitionEofReachedEventHandler(TopicPartitionOffset topicPartitionOffset);
 
-    internal sealed class TopicSubscription : IAutoSubscription
+    public interface ISubscription {
+        void AssignConsumer<K, V>(IConsumer<K, V> consumer);
+
+        void InvokeTopicPartitionEofReached(TopicPartitionOffset topicPartitionOffset);
+        event TopicPartitionEofReachedEventHandler TopicPartitionEofReached;
+    }
+
+    public abstract class Subscription : ISubscription
+    {
+        public event TopicPartitionEofReachedEventHandler TopicPartitionEofReached;
```

Comment:
I don't expect this to be a good idea. One of the specific reasons for akka streams is to fix the burden of working with events.

Reply:
The EOF is of value when you need to know if the data in the actors are up to date. It depends on the reason someone is using Kafka off course.

How would you get the EOF? Is there a way to provide a secondary out-stream, like the broadcaster?